hibernate-core/src/main/java/org/hibernate/cfg/annotations/QueryHintDefinition.java

```diff
@@ -84,6 +84,14 @@ public LockMode getLockMode(String query) {
 		}
 	}
 
+	public Boolean getPassDistinctThrough(String query, String hintName) {
+		String value =(String)  hintsMap.get( hintName );
+		if ( value == null ) {
+			return null;
+		}
+		return getBoolean(query, hintName);
+	}
+
```

Comment:
Seems the only additional value of the above method is the possibility to return `null`, compared with the following one. Seems the method is a generic one (not restricted to 'passDistinctThrough' for the hintName has been one parameter, not hardcoded).

Reply:
Invoking `boolean getBoolean()` brings about code duplication. There might be other more elegant way to achieve the same goal.

Reply:
Thanks for the comment. passDistinctThrough method indeed looked like generic, refactor mistake. I refactored the other part to with Optionals, did not see any use of them though in the project. Is it forbidden to use them ? If so, could you explain why ?

Reply:
Yeah, I have no idea why `Optional` is seldom used. 
Guess you can ask at https://hibernate.zulipchat.com/#narrow/stream/132094-hibernate-orm-dev?

As long as you use `Optional` in private method, it seems safe to use it as you have done.  Maybe the method name is not consistent with existing stuff and even a little bit confusing. Maybe `doGetBoolean`?

Reply:
I asked the question for you there. BTW, it is always a good idea to use zulip for that is the IM tool they used.

Reply:
ok thanks a lot, renamed the method

Reply:
Thanks. BTW, I am a contributor just like you, not team member. Given this PR is not in the PR list of the next release, guess it will take a while for it to be reviewed by team member and merged. Just FYI.